business/apps.go

```diff
@@ -109,78 +134,86 @@ func (in *AppService) GetAppList(ctx context.Context, criteria AppCriteria) (mod
-					svcReferences = append(svcReferences, &ref)
+	for _, clusterApps := range allApps {
+		for keyApp, valueApp := range clusterApps {
+			appItem := &models.AppListItem{
+				Name:         keyApp,
+				IstioSidecar: true,
+				Health:       models.EmptyAppHealth(),
+			}
+			applabels := make(map[string][]string)
+			svcReferences := make([]*models.IstioValidationKey, 0)
+			for _, srv := range valueApp.Services {
+				joinMap(applabels, srv.Labels)
+				if criteria.IncludeIstioResources {
+					vsFiltered := kubernetes.FilterVirtualServicesByService(istioConfigList.VirtualServices, srv.Namespace, srv.Name)
+					for _, v := range vsFiltered {
+						ref := models.BuildKey(v.Kind, v.Name, v.Namespace)
+						svcReferences = append(svcReferences, &ref)
+					}
+					drFiltered := kubernetes.FilterDestinationRulesByService(istioConfigList.DestinationRules, srv.Namespace, srv.Name)
+					for _, d := range drFiltered {
+						ref := models.BuildKey(d.Kind, d.Name, d.Namespace)
+						svcReferences = append(svcReferences, &ref)
+					}
+					gwFiltered := kubernetes.FilterGatewaysByVirtualServices(istioConfigList.Gateways, istioConfigList.VirtualServices)
+					for _, g := range gwFiltered {
+						ref := models.BuildKey(g.Kind, g.Name, g.Namespace)
+						svcReferences = append(svcReferences, &ref)
+					}
+
 				}
 
 			}
 
-		}
-
-		wkdReferences := make([]*models.IstioValidationKey, 0)
-		for _, wrk := range valueApp.Workloads {
-			joinMap(applabels, wrk.Labels)
-			if criteria.IncludeIstioResources {
-				wSelector := labels.Set(wrk.Labels).AsSelector().String()
-				wkdReferences = append(wkdReferences, FilterWorkloadReferences(wSelector, istioConfigList)...)
+			wkdReferences := make([]*models.IstioValidationKey, 0)
+			for _, wrk := range valueApp.Workloads {
+				joinMap(applabels, wrk.Labels)
+				if criteria.IncludeIstioResources {
+					wSelector := labels.Set(wrk.Labels).AsSelector().String()
+					wkdReferences = append(wkdReferences, FilterWorkloadReferences(wSelector, istioConfigList)...)
+				}
 			}
-		}
-		appItem.Labels = buildFinalLabels(applabels)
-		appItem.IstioReferences = FilterUniqueIstioReferences(append(svcReferences, wkdReferences...))
+			appItem.Labels = buildFinalLabels(applabels)
+			appItem.IstioReferences = FilterUniqueIstioReferences(append(svcReferences, wkdReferences...))
 
-		for _, w := range valueApp.Workloads {
-			if appItem.IstioSidecar = w.IstioSidecar; !appItem.IstioSidecar {
-				break
+			for _, w := range valueApp.Workloads {
+				if appItem.IstioSidecar = w.IstioSidecar; !appItem.IstioSidecar {
+					break
+				}
 			}
-		}
-		if criteria.IncludeHealth {
-			appItem.Health, err = in.businessLayer.Health.GetAppHealth(ctx, criteria.Namespace, appItem.Name, criteria.RateInterval, criteria.QueryTime, valueApp)
-			if err != nil {
-				log.Errorf("Error fetching Health in namespace %s for app %s: %s", criteria.Namespace, appItem.Name, err)
+			if criteria.IncludeHealth {
```

Comment:
I was thinking if there is any work related to mc in the health computation, I noticed that there some queries to get the traffic rate that is used in the health, but I don't think we need to include the cluster value in the queries, but I'm not sure. I think that we need to check any inbound and outbound connection to the specific application (from the application's cluster or other clusters too).